core-toolbar.html

```diff
@@ -65,91 +55,135 @@
 -->
 
 <link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../layout/layout.html">
 
-<polymer-element name="core-toolbar" attributes="justify middleJustify bottomJustify">
-<template>
+<dom-module id="core-toolbar">
 
   <link rel="stylesheet" href="core-toolbar.css">
 
-  <div id="bottomBar" class="toolbar-tools" center horizontal layout>
-    <content select=".bottom"></content>
-  </div>
+  <template>
+
+    <div id="bottomBar" class-name="[[bottomBarClassName]]">
+      <content select=".bottom"></content>
+    </div>
+
+    <div id="middleBar" class-name="[[middleBarClassName]]">
+      <content select=".middle"></content>
+    </div>
 
-  <div id="middleBar" class="toolbar-tools" center horizontal layout>
-    <content select=".middle"></content>
-  </div>
+    <div id="topBar" class-name="[[topBarClassName]]">
+      <content></content>
+    </div>
 
-  <div id="topBar" class="toolbar-tools" center horizontal layout>
-    <content></content>
-  </div>
+  </template>
+
+</dom-module>
 
-</template>
 <script>
 
-(function() {
+  (function() {
 
-  Polymer('core-toolbar', {
-    
-    /**
-     * Controls how the items are aligned horizontally.
-     * Options are `start`, `center`, `end`, `between` and `around`.
-     *
-     * @attribute justify
-     * @type string
-     * @default ''
-     */
-    justify: '',
-    
-    /**
-     * Controls how the items are aligned horizontally when they are placed
-     * in the middle.
-     * Options are `start`, `center`, `end`, `between` and `around`.
-     *
-     * @attribute middleJustify
-     * @type string
-     * @default ''
-     */
-    middleJustify: '',
-    
-    /**
-     * Controls how the items are aligned horizontally when they are placed
-     * at the bottom.
-     * Options are `start`, `center`, `end`, `between` and `around`.
-     *
-     * @attribute bottomJustify
-     * @type string
-     * @default ''
-     */
-    bottomJustify: '',
-    
-    justifyChanged: function(old) {
-      this.updateBarJustify(this.$.topBar, this.justify, old);
-    },
-    
-    middleJustifyChanged: function(old) {
-      this.updateBarJustify(this.$.middleBar, this.middleJustify, old);
-    },
-    
-    bottomJustifyChanged: function(old) {
-      this.updateBarJustify(this.$.bottomBar, this.bottomJustify, old);
-    },
-    
-    updateBarJustify: function(bar, justify, old) {
-      if (old) {
-        bar.removeAttribute(this.toLayoutAttrName(old));
+    'use strict';
+
+    function classNames(obj) {
+      var classNames = [];
+      for (var key in obj) {
+        if (obj.hasOwnProperty(key) && obj[key]) {
+          classNames.push(key);
+        }
       }
-      if (justify) {
-        bar.setAttribute(this.toLayoutAttrName(justify), '');
+
+      return classNames.join(' ');
+    }
+
+    function calcBarClassName(barJustify) {
```

Comment:
I originally added it to the object passed to `Polymer`, but it didn't reference `this` at all. I remember a discussion about class functions instead of instance functions, and that's what this should be. Do you know where those should live?